### PR TITLE
Use macrovich to do macrology.

### LINKED ISSUE #13: Compilation warnings with ClojureScript and shadow-cljs
Given:

```clojure
{:deps {org.clojure/clojurescript {:mvn/version "1.10.773"}
        thheller/shadow-cljs {:mvn/version "2.11.15"}
        exoscale/coax {:mvn/version "1.0.0-alpha11"}}}
```

shadow-cljs reports:

```
------ WARNING #1 - :undeclared-var --------------------------------------------
 Resource: exoscale/coax/coercer.cljc:23:4
 Can't take value of macro exoscale.coax.coercer/invalid-on-throw!
--------------------------------------------------------------------------------

------ WARNING #2 - :undeclared-var --------------------------------------------
 Resource: exoscale/coax/coercer.cljc:35:4
 Can't take value of macro exoscale.coax.coercer/invalid-on-throw!
--------------------------------------------------------------------------------

------ WARNING #3 - :undeclared-var --------------------------------------------
 Resource: exoscale/coax/coercer.cljc:69:6
 Can't take value of macro exoscale.coax.coercer/invalid-on-throw!
--------------------------------------------------------------------------------

------ WARNING #4 - :undeclared-var --------------------------------------------
 Resource: exoscale/coax/coercer.cljc:80:6
 Can't take value of macro exoscale.coax.coercer/invalid-on-throw!
--------------------------------------------------------------------------------
```

I haven't yet managed to make a minimal repro, so I don't know what exactly is the problem. I can look into it, but I just wanted to create the issue so that I don't forget.

All I use are `coax/def` and `coax/coerce`, so it's either of those.

### PR DESCRIPTION
**Pull request by jellelicht**
This seems to fix #13 for me

### CODE REVIEW

**Comment by mpenet** [comment]
Hi, 

Thanks for looking into this, I have to admit I never spent time to check the cljs side of things in detail. 

Would there be a way to achieve the same without using macrovich? I'd like to avoid if possible to add a dependency on it.

**Comment by mpenet** [comment]
I am not sure, but I think just moving this macro in a separate/dedicated ns (still in cljc) and calling either require or require-macro on it depending on the context from coercer.cljc would fix the issue.  Could you confirm this?

**Comment by mpenet** [comment]
Something like that: https://github.com/exoscale/coax/commit/6b14047d4417504554a3853b21913b63f6654c54

**Comment by jellelicht** [comment]
It actually does not work properly:
``` Use of undeclared Var exoscale.coax.coercer/java```
I think this happens because the reader conditional in the body of the invalid-on-throw! actually is expanded on the clojure side of things. This is _exactly_ one or the problems that macrovich solves, as per their 
Observe in `exoscale.coax.utils`:
```clojure
(defmacro broken []
  #?(:clj "clojure" :cljs "clojurescript"))
```

and then in `exoscale.coax.coercer`:
```clojure
(defn random-broken []
  (u/broken))
```

and then from my cljs program/repl:
```clojure
(random-broken) ; => "clojure"
```

TLDR: if we want to fix this properly, it should be done in the way that macrovich does it; whether we copy-paste their approach or simply use it as a compile-time dep is of course totally up to you.

**Comment by mpenet** [comment]
I think we'll just adopt macrology then. It's a small enough dependency.

**Comment by mpenet** [comment]
Thanks for the fix, I ll  release a new version shortly

**Comment by mpenet** [comment]
available as 1.0.0-alpha15
